debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.oracle.logminer;
+
+import java.util.List;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.connector.oracle.OracleConnectorConfig;
+import io.debezium.connector.oracle.OracleDatabaseSchema;
+import io.debezium.relational.TableId;
+
+/**
+ * This utility class contains SQL statements to configure, manage and query Oracle LogMiner
+ */
+class SqlUtils {
+
+    private static final String LOGMNR_CONTENTS_VIEW = "V$LOGMNR_CONTENTS";
+    private static final String LOGMNR_AUDIT_TABLE = "LOG_MINING_AUDIT";
+
+    static final String BUILD_DICTIONARY = "BEGIN DBMS_LOGMNR_D.BUILD (options => DBMS_LOGMNR_D.STORE_IN_REDO_LOGS); END;";
+    static final String CURRENT_SCN = "SELECT CURRENT_SCN FROM V$DATABASE";
+    static final String CURRENT_TIMESTAMP = "select current_timestamp from dual";
+    static final String END_LOGMNR = "BEGIN SYS.DBMS_LOGMNR.END_LOGMNR(); END;";
+    static final String OLDEST_FIRST_CHANGE = "SELECT MIN(FIRST_CHANGE#) FROM V$LOG";
+    static final String ALL_ONLINE_LOGS = "SELECT MIN(F.MEMBER) AS FILE_NAME, L.NEXT_CHANGE# AS NEXT_CHANGE, F.GROUP# " +
+            " FROM V$LOG L, V$LOGFILE F " +
+            " WHERE F.GROUP# = L.GROUP# AND L.NEXT_CHANGE# > 0 " +
+            " GROUP BY F.GROUP#, L.NEXT_CHANGE# ORDER BY 3";
+
+    static final String REDO_LOGS_STATUS = "SELECT F.MEMBER, R.STATUS FROM V$LOGFILE F, V$LOG R WHERE F.GROUP# = R.GROUP# ORDER BY 2";
+    static final String SWITCH_HISTORY_TOTAL_COUNT = "select 'total', count(1) from v$archived_log where first_time > trunc(sysdate)" +
+            " and dest_id = (select dest_id from V$ARCHIVE_DEST_STATUS where status='VALID' and type='LOCAL')";
+    static final String CURRENT_REDO_LOG_NAME = "select f.member from v$log log, v$logfile f  where log.group#=f.group# and log.status='CURRENT'";
```

Comment:
You only need one. If there is a group just take one of the files in the group.
```
        val CURRENT_REDO_FILENAME = """
            SELECT DISTINCT(first_value(lf.MEMBER) OVER (PARTITION BY lf.GROUP# ORDER BY lf.MEMBER)) AS FILENAME,
                   FIRST_CHANGE# AS FIRST,
                   NEXT_CHANGE# - 1 AS LAST
            FROM V$LOGFILE lf, V$LOG l
            WHERE lf.GROUP# = l.GROUP# AND l.STATUS = 'CURRENT'
        """.trimIndent()
```

Reply:
@AndreyIg any insight on this suggestion?

Reply:
it would be more than one in RAC environment

Reply:
Ok.

Reply:
I can't find that the archive_logs are used. This is the query I use to get all log files. Archived and online combined.
```
            SELECT *
            FROM (
                 SELECT DISTINCT(first_value(lf.MEMBER) OVER (PARTITION BY lf.GROUP# ORDER BY lf.MEMBER)) AS FILENAME,
                        FIRST_CHANGE# AS FIRST,
                        NEXT_CHANGE# - 1 AS LAST
                 FROM V$LOGFILE lf, V$LOG l
                 WHERE lf.GROUP# = l.GROUP# AND l.STATUS = 'CURRENT'
                 UNION
                 SELECT NAME AS FILENAME, 
                        FIRST_CHANGE# AS FIRST, 
                        NEXT_CHANGE# - 1 AS LAST
                 FROM V$ARCHIVED_LOG
                 WHERE DELETED = 'NO' AND STANDBY_DEST = 'NO'
             )
            WHERE ? <= LAST
```

Reply:
Nice query. You are filtering out duplication later, right?

Reply:
I did not see any duplicates up to today actually. It only happened once. Was that because of rotation and the having some overlapping states? But indeed. I have to filter for duplicates and I'll do that in this query actually.

Reply:
`STANDBY_DEST = 'NO'` seems to be needed. Otherwise the filenames can have weird values. I'm no expert on Oracle so this for me is trial and error. Fail fast, fix forward :)

Reply:
@Naros follow-up for this too?

Reply:
Logged as [DBZ-2529](https://issues.redhat.com/browse/DBZ-2529).